src/js/controller.js

```diff
@@ -0,0 +1,70 @@
+import { Address } from './single-address';
+import { setInitialFormValues } from './helpers';
+
+export class Controller {
+
+  constructor(store, view) {
+    this.store = store;
+    this.view = view;
+    
+    this.initialFormState = setInitialFormValues();
+
+    this.addressList = this.store.fetchAddressList();
+
+    this.view.bindAddAddress(this.addNewAddress.bind(this));
+    this.view.bindRemoveAddress(this.removeAddress.bind(this));
+    this.view.bindEditAddress(this.editAddress.bind(this));
+    this.view.bindSaveEditedAddress(this.saveEditedAddress.bind(this));
+    this.view.bindCancelEditAddress(this.cancelEditAddress.bind(this));
+  };
+
+  addNewAddress() {
+    const addressConfig = this.view.getFormInputsValues();
+    const list = this.store.fetchAddressList();
+    const newAddress = new Address(addressConfig);
+    const newList = [ ...list, newAddress];
+
+    this.store.saveAddressList(newList);
+    return newAddress;
+  };
+
+  removeAddress(id) {
+    const addressList = this.store.fetchAddressList();
+    const newList = addressList.filter(address => address.id !== id);
+    this.store.saveAddressList(newList);
```

Comment:
Ten sam błąd. Powinieneś mieć po prostu `this.store.removeAddress(id)`. Tak jak mówiłem kontroler to tylko most między V i M, przekazuje instrukcje i dane z jednego do drugiego i na odwrót.

Reply:
Pod tym samym kątem sprawdź inne metody.

Reply:
To kluczowe bo w Angularze też masz podział odpowiedzialności na kontrolery i serwisy.